File: scala/sdktest/ConvertEntityTest.scala
// Generated basic-flow test for the convert entity (model-driven;
// mirrors the java TestEntity generator). A dependency-free scala-cli test
// object driven by SdkEntityTestMain. Runs against the in-memory test
// transport seeded with the shipped ConvertTestData.json fixtures.

import java.util.{ArrayList, LinkedHashMap, List => JList, Map => JMap}

import voxgig.univecsdk.core.{Helpers, UnivecSDK}
import voxgig.univecsdk.utility.struct.Struct

object ConvertEntityTest {

  def run(rep: SdkTestReport): Unit = {
    rep.scope("convert.instance") {
      val testsdk = UnivecSDK.testSDK()
      val ent = testsdk.convert(null)
      rep.check("convert.instance", ent != null, "expected non-null convert entity")
    }

    rep.scope("convert.basic") {
      val entityData = Helpers.toMapAny(SdkTestSupport.readJson(
          "../.sdk/test/entity/convert/ConvertTestData.json"))
      val options = new LinkedHashMap[String, Object]()
      options.put("entity", entityData.get("existing"))
      val client = UnivecSDK.testSDK(options, null)

      val idmap = new LinkedHashMap[String, Object]()
      idmap.put("convert01", "CONVERT01")
      idmap.put("convert02", "CONVERT02")
      idmap.put("convert03", "CONVERT03")
      val now = System.currentTimeMillis()

      // CREATE
      val convertRef01Ent = client.convert(null)
      var convertRef01Data = Helpers.toMapAny(Struct.getprop(
          Struct.getpath(entityData, "new.convert"), "convert_ref01"))
      val convertRef01DataResult = convertRef01Ent.create(convertRef01Data, null)
      convertRef01Data = Helpers.toMapAny(convertRef01DataResult match { case e: SdkEntity => e.data(); case o => o })
      rep.check("convert.create.map", convertRef01Data != null, "expected create result to be a map")
    }
  }
}

File: elixir/test/convert_entity_test.exs
# Convert entity test (offline, mock transport)

defmodule Univec.ConvertEntityTest do
  use ExUnit.Case

  alias Voxgig.Struct, as: S
  alias Univec.Helpers, as: H
  alias Univec.Json

  defp fixture do
    Json.parse(File.read!("../.sdk/test/entity/convert/ConvertTestData.json"))
  end

  defp mk_sdk do
    existing = H.or_(S.getpath(fixture(), "existing"), S.jm([]))
    Univec.test(S.jm(["entity", existing]))
  end

  defp first_id do
    existing = H.or_(S.getpath(fixture(), "existing.convert"), S.jm([]))
    keys = S.keysof(existing)
    if keys == [], do: nil, else: hd(keys)
  end

  test "should create instance" do
    sdk = Univec.test()
    ent = Univec.convert(sdk)
    assert ent != nil
  end

  test "should create then read back" do
    sdk = Univec.test(S.jm(["entity", S.jm(["convert", S.jm([])])]))
    ent = Univec.convert(sdk)
    created = Univec.Entity.Convert.create(ent, S.jm(["name", "test-create"]))
    made = Univec.EntityBase.data_get(created)
    assert S.ismap(made)
    assert S.getprop(made, "id") != nil
  end
end
